back/app/controllers/web_api/v1/internal_comments_controller.rb

```diff
@@ -0,0 +1,186 @@
+      .order(:created_at)
+      .includes(*include_attrs)
+
+    # We're doing this merge in ruby, since the combination of a self-join
+    # with different sorting criteria per depth became tremendously complex in
+    # one SQL query
+    @comments = merge_comments(root_comments.to_a, child_comments.to_a)
+
+    serialization_options = { params: jsonapi_serializer_params, include: [:author] }
+
+    render json: {
+      **WebApi::V1::InternalCommentSerializer.new(@comments, serialization_options).serializable_hash,
+      links: page_links(root_comments)
+    }
+  end
+
+  def children
+    @comments = policy_scope(InternalComment, policy_scope_class: @policy_class::Scope)
+      .where(parent: params[:id])
+      .order(:lft)
+    @comments = paginate @comments
+
+    serialization_options = { params: jsonapi_serializer_params, include: [:author] }
+
+    render json: linked_json(@comments, WebApi::V1::InternalCommentSerializer, serialization_options)
+  end
+
+  def show
+    render json: WebApi::V1::InternalCommentSerializer.new(
+      @comment,
+      params: jsonapi_serializer_params,
+      include: [:author]
+    ).serializable_hash
+  end
+
+  def create
+    @comment = InternalComment.new comment_create_params
+    @comment.post_type = @post_type
+    @comment.post_id = @post_id
+    @comment.author ||= current_user
+    authorize @comment, policy_class: @policy_class
+
+    SideFxInternalCommentService.new.before_create @comment, current_user
+    if @comment.save
+      SideFxInternalCommentService.new.after_create @comment, current_user
+      render json: WebApi::V1::InternalCommentSerializer.new(
+        @comment,
+        params: jsonapi_serializer_params,
+        include: [:author]
+      ).serializable_hash, status: :created
+    else
+      render json: { errors: @comment.errors.details }, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    @comment.attributes = comment_update_params
+    # We cannot pass policy class to permitted_attributes
+    # @comment.attributes = pundit_params_for(@comment).permit(@policy_class.new(current_user, @comment).permitted_attributes_for_update)
+    authorize @comment, policy_class: @policy_class
+
+    SideFxInternalCommentService.new.before_update @comment, current_user
+    if @comment.save
+      SideFxInternalCommentService.new.after_update @comment, current_user
+      render json: WebApi::V1::InternalCommentSerializer.new(
+        @comment,
+        params: jsonapi_serializer_params,
+        include: [:author]
+      ).serializable_hash, status: :ok
+    else
+      render json: { errors: @comment.errors.details }, status: :unprocessable_entity
+    end
+  end
+
+  def mark_as_deleted
+    return unless @comment.author_id == current_user&.id
+
+    @comment.publication_status = 'deleted'
+    if @comment.save
+      SideFxInternalCommentService.new.after_mark_as_deleted(@comment, current_user)
+      head :accepted
```

Comment:
I'd rather use:
```suggestion
      head :no_content
```
or return the updated comment with 200.

Accepted is usually used for async processing.

Reply:
✅ Done